File: set2/python/p10.py
import utils

def main():
    """Main."""
    with open('p10_problem.txt', 'r') as f:
        data = f.read()
        code = bytearray(data.decode('base64'))

    key = 'YELLOW SUBMARINE'
    iv = bytearray([0] * len(key))

    decrypted = utils.apply_CBC('decrypt', code, key, iv)
    print ''.join([chr(b) for b in decrypted])
    print 'p10 ok'

if __name__ == '__main__':
    main()

File: set2/rust/src/p10.rs
use openssl::symm::Cipher;
use rust_utils::*;
use std::fs;

/*----------------------------------------------------------------------------*/

pub fn main() {
    let data = fs::read_to_string("data/p10.txt")
        .expect("file read failed")
        .replace("\n", "");

    let decoded = to_bytes::from_b64(&data);
    let key = to_bytes::from_utf8("YELLOW SUBMARINE");
    let iv = vec![0].repeat(16);
    let decrypted = aes::decrypt_cbc(Cipher::aes_128_ecb(), 16, &decoded, &key, &iv);
    match decrypted {
        Ok(msg) => println!("{:?}", from_bytes::to_utf8(&msg)),
        Err(err) => println!("{}", err),
    }
}
